waddrmgr/manager.go

```diff
@@ -1470,6 +1485,188 @@ func (m *Manager) LastInternalAddress(account uint32) (ManagedAddress, error) {
 	return acctInfo.lastInternalAddr, nil
 }
 
+// NewAccount creates and returns a new account stored in the manager based
+// on the given account name.  If an account with the same name already exists,
+// ErrDuplicateAccount will be returned.  Since creating a new account requires
+// access to the cointype keys (from which extended account keys are derived),
+// it requires the manager to be unlocked.
+func (m *Manager) NewAccount(name string) (uint32, error) {
+	var account uint32
+	var coinTypePrivEnc []byte
+
+	// Validate account name
+	if err := ValidateAccountName(name); err != nil {
+		return account, err
+	}
+
+	// Check that account with the same name does not exist
+	_, err := m.LookupAccount(name)
+	if err == nil {
+		str := fmt.Sprintf("account with the same name already exists")
+		return account, managerError(ErrDuplicateAccount, str, err)
+	}
+
+	// Fetch latest account, and create a new account in the same transaction
+	err = m.namespace.Update(func(tx walletdb.Tx) error {
+		var err error
+		// Fetch the latest account number to generate the next account number
+		account, err = fetchLastAccount(tx)
+		if err != nil {
+			return err
+		}
+		// TODO: needs to be atomic?
+		account++
+		// Fetch the cointype key which will be used to derive the next account extended keys
+		_, coinTypePrivEnc, err = fetchCoinTypeKeys(tx)
+		if err != nil {
+			return err
+		}
+
+		// Decrypt the cointype key
+		serializedKeyPriv, err := m.cryptoKeyPriv.Decrypt(coinTypePrivEnc)
+		if err != nil {
+			str := fmt.Sprintf("failed to decrypt cointype serialized private key")
+			return managerError(ErrLocked, str, err)
+		}
+		coinTypeKeyPriv, err := hdkeychain.NewKeyFromString(string(serializedKeyPriv))
+		if err != nil {
+			str := fmt.Sprintf("failed to create cointype extended private key")
+			return managerError(ErrKeyChain, str, err)
+		}
+
+		// Derive the account key using the cointype key
+		acctKeyPriv, err := deriveAccountKey(coinTypeKeyPriv, account)
```

Comment:
The coin type extended key bytes need to be zeroed as soon as they are used.
```Go
coinTypeKeyPriv.Zero()
```

NOTE: This is intentionally before the error check so it happens regardless.